src/Host.cpp

```diff
@@ -3903,6 +4014,48 @@ void Host::createMapper(const bool loadDefaultMap)
     raiseEvent(mapOpenEvent);
 }
 
+void Host::showHideOrCreateVideoPlayer()
+{
+    if (mpMedia->mpVideoPlayer) {
+        toggleVideoPlayerVisibility();
+        return;
+    }
+
+    createVideoPlayer();
+}
+
+void Host::toggleVideoPlayerVisibility()
+{
+    bool visStatus = mpMedia->mpVideoPlayer->isVisible();
```

Comment:
Probably need some `nullptr` checks:
```suggestion
    if (!mpMedia || !mpMedia->mpVideoPlayer) {
        return;
    }
    bool visStatus = mpMedia->mpVideoPlayer->isVisible();
```